hollow-jsonadapter/src/main/java/com/netflix/hollow/jsonadapter/HollowJsonAdapter.java

```diff
@@ -510,6 +519,14 @@ private void skipObjectField(JsonParser parser, JsonToken token) throws IOExcept
         }
     }
 
+    private int addRecord(String type, HollowWriteRecord rec, FlatRecordWriter flatRecordWriter) {
+        if(flatRecordWriter != null) {
```

Comment:
There's a long call-chain of methods that are just passing `flatRecordWriter` along to the next method. I think that's a bit of a design smell. But I think that can be addressed in a separate refactoring.

Reply:
Yes, i agree with this, but i don't currently know what a better solution might be.